src/main/java/org/gridsuite/boundary/importer/job/BoundaryAcquisitionJob.java

```diff
@@ -0,0 +1,155 @@
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+
+/**
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+public final class BoundaryAcquisitionJob {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BoundaryAcquisitionJob.class);
+
+    private BoundaryAcquisitionJob() {
+    }
+
+    private static void importBoundary(byte[] boundaryContent,
+                                       List<String> allBoundaryIds,
+                                       String fileName,
+                                       CgmesBoundaryServiceRequester cgmesBoundaryServiceRequester,
+                                       List<String> filesImported,
+                                       List<String> filesAlreadyImported,
+                                       List<String> filesImportFailed) throws InterruptedException {
+        try (Reader reader = new InputStreamReader(new ByteArrayInputStream(boundaryContent))) {
+            // parse full model to get the id
+            FullModel fullModel = FullModel.parse(reader);
+            String id = fullModel.getId();
+
+            if (!allBoundaryIds.contains(id)) {
+                // import the boundary to the cgmes boundary server
+                LOGGER.info("Importing boundary file '{}'...", fileName);
+
+                boolean importOk = cgmesBoundaryServiceRequester.importBoundary(new TransferableFile(fileName, boundaryContent));
+                if (importOk) {
+                    filesImported.add(fileName);
+                } else {
+                    filesImportFailed.add(fileName);
+                }
+            } else {
+                filesAlreadyImported.add(fileName);
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        } catch (InterruptedException e) {
+            throw e;
+        }
+    }
+
+    private static void handleZipBoundaryContainer(TransferableFile acquiredFile,
+                                                   List<String> allBoundaryIds,
+                                                   CgmesBoundaryServiceRequester cgmesBoundaryServiceRequester,
+                                                   List<String> filesImported,
+                                                   List<String> filesAlreadyImported,
+                                                   List<String> filesImportFailed) throws InterruptedException {
+        String fileName;
+        try (SecuredZipInputStream zis = new SecuredZipInputStream(new ByteArrayInputStream(acquiredFile.getData()), CgmesBoundaryUtils.MAX_ZIP_ENTRIES_COUNT, CgmesBoundaryUtils.MAX_ZIP_SIZE)) {
+            ZipEntry entry = zis.getNextEntry();
+            while (entry != null) {
+                if (new File(entry.getName()).getCanonicalPath().contains("..")) {
```

Comment:
A file or directory can be named with more than one '.'  'like..this...' for example isn't  startsWith better ?

Reply:
Done